pkg/services/user/identity.go

```diff
@@ -105,17 +105,15 @@ func (u *SignedInUser) HasUniqueId() bool {
 
 // GetCacheKey returns a unique key for the entity.
 // Add an extra prefix to avoid collisions with other caches
-func (u *SignedInUser) GetCacheKey() (string, error) {
-	if u.IsRealUser() {
-		return fmt.Sprintf("%d-user-%d", u.OrgID, u.UserID), nil
+func (u *SignedInUser) GetCacheKey() string {
+	namespace, id := u.GetNamespacedID()
+	if !u.HasUniqueId() {
```

Comment:
We could double proof:
```suggestion
	if !u.HasUniqueId() || id == "0" {
```

Reply:
This is already covered by has unique id